src/diffusers/loaders/single_file_utils.py

```diff
@@ -278,6 +279,12 @@ def infer_original_config_file(class_name, checkpoint):
     elif class_name == "ControlNetModel":
         config_url = CONFIG_URLS["controlnet"]
 
+    elif len(checkpoint) == 512:
+        config_url = CONFIG_URLS["test_single_file_sd"]
+    
+    elif len(checkpoint) == 701:
+        config_url = CONFIG_URLS["test_single_file_sdxl"]
```

Comment:
I don't think there's any other sane way for us to verify this.

Reply:
Can't we add this into the test module? Why have it in `src` if it's only relevant to tests? We can pass it in using `Pipeline.from_single_file(original_config_file="<file url>")`. `original_config_file` can be a URL now.